### PR TITLE
comments/possible changes

### PR DESCRIPTION
**Pull request by blairconrad**
First off, I should warn you that I'm completely unfamiliar with Cecil. Also IL. And Fody. So any comments may be way off!

Second, the code was a pleasure to read. I understand there are bits still to come, but I liked what I saw. My approach was to read the README.md, then look at the sample Assembly to weave, and those two things gave me a great view. It was easy to picture how the assembly would be used and what would be done to it. 

Then as suggested, I read `ModuleWeaver` and everything just fell into place. It was like reading a story.

### CODE REVIEW

File: EnableFaking.Fody.Test/WeaverHelper.cs [hunk 1]
```diff
@@ -20,12 +20,9 @@ public static Assembly WeaveAssembly()
             File.Copy(assemblyPath, newAssembly, true);
 
             var moduleDefinition = ModuleDefinition.ReadModule(newAssembly);
-            var weavingTask = new ModuleWeaver
-            {
-                ModuleDefinition = moduleDefinition
-            };
+            var weavingTask = new ModuleWeaver();
 
-            weavingTask.Execute();
```
Comment (blairconrad) [inline EnableFaking.Fody.Test/WeaverHelper.cs:28]: Probably best to ignore this and focus on ModuleWeaver.cs

File: EnableFaking.Fody/ModuleWeaver.cs [hunk 2]
```diff
@@ -31,17 +31,21 @@ public ModuleWeaver()
...
-
+        // ModuleDefinition's only used by Execute, so why not pass it in?
+        // Makes it easy to weave more than one module (not sure if that makes sense).
+        // Also, don't enjoy classes that aren't complete when constructed, but 
+        // that's my bias!
         public Action<string> LogInfo { get; set; }
 
-        public void Execute()
+
+        // Execute seemed kind of generic. Why not Weave? Unless Execute's a convention�
```
Comment (blairconrad) [inline EnableFaking.Fody/ModuleWeaver.cs:41]: Question-mark-in-diamond was an ellipsis when it left my keyboard. _blush_

File: EnableFaking.Fody.Test/WeaverHelper.cs (hunk 1 above)
Comment (philippdolder) [inline EnableFaking.Fody.Test/WeaverHelper.cs:28]: Not sure yet if we really need `WeaverHelper`

**Comment by philippdolder** [comment]
Thanks for the compliments regarding the readability and understandability of the code

**Comment by blairconrad** [comment]
Regarding Execute/ModuleDefinition - sorry. Didn't mean to waste your time with (ignorance of Fody + laziness). I shoulda done more research.

**Comment by blairconrad** [comment]
Oh, and for the comments on readability - sure. I often have to read legacy stuff at work that just hurts my head, so why not praise where it's deserved!

Also, I don't really have many other comments of worth. I've never used a Fody plugin, and feel like I don't quite have a handle on how they're supposed to be used. I think I need to go reread the docs and/or look for a tutorial.
And I've clearly demonstrated my ignorance on how Fody calls the plugins! :smile:

**Pull request closed without merging**